src/txn_builder.rs

```diff
@@ -1469,37 +1539,83 @@ impl<L: Ledger> TransactionState<L> {
             current_amount += record.ro.amount;
             result.push((record.ro.clone(), record.uid));
             if current_amount >= amount {
-                return Ok((result, current_amount - amount));
+                return Ok((result, (current_amount - amount) as i64));
+            }
+        }
+
+        if allow_insufficient {
+            Ok((result, (current_amount - amount) as i64))
+        } else {
+            Err(TransactionError::InsufficientBalance {
+                asset: *asset,
+                required: amount,
+                actual: current_amount,
+            })
+        }
+    }
+
+    #[allow(clippy::type_complexity)]
+    fn find_records(
+        &self,
+        asset: &AssetCode,
+        owner_key_pairs: &[UserKeyPair],
+        frozen: FreezeFlag,
+        amount: u64,
+        max_records: Option<usize>,
```

Comment:
This function appears to no longer respect `max_records`. Each individual call to `find_records_with_pub_key` will respect it, but overall we might return as many as `max_records*owner_key_pairs.len()` records, if I'm not mistaken. This will cause problems with `find_fee_input`